pkg/auto/lights/auto_test.go

```diff
@@ -30,59 +33,69 @@ func TestPirsTurnLightsOn(t *testing.T) {
 
 	testActions := newTestActions(t)
 
-	now := time.Unix(0, 0)
+	clock := newTestClock(t)
 
-	automation := PirsTurnLightsOn(rootNode, zap.NewNop())
+	loggerConfig := zap.NewDevelopmentConfig()
+	loggerConfig.DisableStacktrace = true
+	logger, _ := loggerConfig.Build()                                          // the test is sometimes failing, this might help with debugging
+	logger = logger.Named(fmt.Sprintf("%x", math.Float64bits(rand.Float64()))) // helps with --count>1 tests
```

Comment:
What's the motivation for this random-float-to-int conversion?
Seems like it just gets a random integer with a certain limited range?
Also, `x/exp/rand` is used here which is deprecated.
Couldn't [rand.Uint64](https://pkg.go.dev/math/rand/v2#Uint64) be used instead?

Reply:
Good catch, and you are correct. I went down the road of thinking _I want 32 or 64 bits of random data_ and got caught in the wrong are of code.